restconnector/RestConnector.h

```diff
@@ -11,20 +11,35 @@
 namespace restconnector
 {
 
+/**
+ * The RestConnector class is responsible starting a HTTP server
+ * in a dedicated thread, according to a given hostname and port.
+ * HTTP requests are forwarded to an implicitely registered handler.
+ */
 class RestConnector
 {
 public:
 
-    RestConnector( const std::string& address, const std::string& port );
+    /**
+     * Default constructor.
+     * @param hostname Hostname or IP address
+     * @param port Listening port
+     */
+    RestConnector( const std::string& hostname, const std::string& port );
```

Comment:
Following that line of thought then you should merge the hostname and the port into a single string (as 0MQ does, for example) and let the library do all the parsing. Then you don't need separate command line options for hostname and port right and you can let RestConnector split the string in two, right?

I'm with Daniel in this one. The fact that you use Cppnetlib is an implementation detail and that should not influence the semantics of the API. Converting a string into an integer is a line of code and even is RestConnector does it, the client still has to check the error, I guess with exception handling as this is the constructor, in which case adding a boost::lexical_cast is not a big deal in the user side.
So I'd vote for either a uint16_t for the port, or a single string for both things, but not this half-in-between solution.

A side comment. Unit tests shouldn't be a planned task, they have to be part of the "definition of done".